libs/execution_context/ebpf_maps.c

```diff
@@ -43,33 +43,126 @@ typedef struct _ebpf_core_object_map
+    (EBPF_LRU_ENTRY_LIST_ENTRY_OFFSET(partition_count) + (partition_count) * sizeof(ebpf_list_entry_t))
+
+/**
+ * @brief Macro to calculate the offset of last used time in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_LAST_USED_TIME_OFFSET(partition_count) \
+    (EBPF_LRU_ENTRY_GENERATION_OFFSET(partition_count) + (partition_count) * sizeof(size_t))
+
+/**
+ * @brief Macro to calculate the offset of key in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_KEY_OFFSET(partition_count) \
+    (EBPF_LRU_ENTRY_LAST_USED_TIME_OFFSET(partition_count) + (partition_count) * sizeof(size_t))
+
+/**
+ * @brief Macro to compute the size of the key history entry.
+ */
+#define EBPF_LRU_ENTRY_SIZE(partition_count, key_size) (EBPF_LRU_ENTRY_KEY_OFFSET(partition_count) + key_size)
+
+/**
+ * @brief Macro to get a pointer to an array of list entries in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_LIST_ENTRY_PTR(map, entry) \
+    ((ebpf_list_entry_t*)(((uint8_t*)entry) + EBPF_LRU_ENTRY_LIST_ENTRY_OFFSET(map->partition_count)))
+
+/**
+ * @brief Macro to get a pointer to an array of generations in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_GENERATION_PTR(map, entry) \
+    ((size_t*)(((uint8_t*)entry) + EBPF_LRU_ENTRY_GENERATION_OFFSET(map->partition_count)))
+
+/**
+ * @brief Macro to get a pointer to an array of last used times in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_LAST_USED_TIME_PTR(map, entry) \
+    ((size_t*)(((uint8_t*)entry) + EBPF_LRU_ENTRY_LAST_USED_TIME_OFFSET(map->partition_count)))
+
+/**
+ * @brief Macro to get a pointer to the key in the key history entry.
+ */
+#define EBPF_LRU_ENTRY_KEY_PTR(map, entry) \
+    ((uint8_t*)(((uint8_t*)entry) + EBPF_LRU_ENTRY_KEY_OFFSET(map->partition_count)))
+
+/**
+ * @brief The partition of the LRU map key history.
+ */
+__declspec(align(EBPF_CACHE_LINE_SIZE)) typedef struct _ebpf_lru_partition
 {
-    ebpf_core_map_t core_map;   //< Core map structure.
     ebpf_list_entry_t hot_list; //< List of ebpf_lru_entry_t containing keys accessed in the current generation.
     ebpf_list_entry_t
         cold_list; //< List of ebpf_lru_entry_t containing keys accessed in previous generations, sorted by generation.
     ebpf_lock_t lock;          //< Lock to protect access to the hot, cold lists, current generation, and hot list size.
     size_t current_generation; //< Current generation. Updated when the hot list is merged into the cold list.
     size_t hot_list_size;      //< Current size of the hot list.
     size_t hot_list_limit;     //< Maximum size of the hot list.
+} ebpf_lru_partition_t;
+
+/**
+ * @brief The map definition for an LRU map.
+ */
+typedef struct _ebpf_core_lru_map
+{
+    ebpf_core_map_t core_map; //< Core map structure.
+    size_t partition_count;   //< Number of LRU partitions. Limited to a maximum of EBPF_LRU_MAXIMUM_PARTITIONS.
+    uint8_t padding[24];      //< Required to ensure partitions are cache aligned.
```

Comment:
I'm still not sure about this hard-coded 24 byte padding. Why not
`DECLSPEC_CACHEALIGN ebpf_lru_partition_t partitions[1];`
to ensure the partitions field is cache aligned on any architecture?

Reply:
Oh, the `ebpf_lru_partition_t` type is already cache aligned. The padding field is redundant, then?